homework_14/homework_14_2.py

```diff
@@ -0,0 +1,125 @@
+                return self.expression(parts[0]) / self.expression(parts[2])
+            if parts[1] == '+':
+                return self.expression(parts[0]) + self.expression(parts[2])
+            if parts[1] == '-':
+                return self.expression(parts[0]) - self.expression(parts[2])
+            if parts[1] == '^':
+                return self.expression(parts[0]) ** self.expression(parts[2])
+        return None
+
+    @staticmethod
+    def summary(a, b):
+        """Return the sum of two numbers."""
+        return f'{a} + {b} = {a + b}'
+
+    @staticmethod
+    def subtraction(a, b):
+        """Return the difference of two numbers."""
+        return f'{a} - {b} = {a - b}'
+
+    @staticmethod
+    def multiplication(a, b):
+        """Return the product of two numbers."""
+        return f'{a} * {b} = {a * b}'
+
+    @staticmethod
+    def division(a, b):
+        """Return the quotient of two numbers, or an error message
+        if dividing by zero."""
+        if b != 0:
+            return f'{a} / {b} = {a / b}'
+        return 'Error! Division by zero!'
+
+    @staticmethod
+    def exponentiation(a, b):
+        """Return the result of raising a number
+         to the power of another number."""
+        return f'{a} ^ {b} = {a ** b}'
+
+    def calc(self):
+        """Perform the calculation based on the selected menu number."""
+        if self.menu_number == 6:
+            input_expr = input('Enter your expression here: ')
+            result = self.expression(input_expr)
+            print('Answer is:', result)
+            return result
+        while True:
+            try:
+                a = Decimal(input('Enter your first number: '))
+                b = Decimal(input('Enter your second number: '))
+            except InvalidOperation:
+                print('Please enter only numbers, thank you')
+                continue
+
+            if self.menu_number == 1:
+                print(self.summary(a, b))
+            elif self.menu_number == 2:
+                print(self.subtraction(a, b))
+            elif self.menu_number == 3:
+                print(self.multiplication(a, b))
+            elif self.menu_number == 4:
+                print(self.division(a, b))
+            elif self.menu_number == 5:
+                print(self.exponentiation(a, b))
+            else:
+                print("Wrong!")
+            break
+        return None
+
+
+calculator = SimpleCalculator()
+calculator.get_menu_number()
+calculator.calc()
+assert calculator.summary(Decimal('3'), Decimal('2')) == '3 + 2 = 5'
+assert calculator.subtraction(Decimal('5'), Decimal('2')) == '5 - 2 = 3'
+assert calculator.multiplication(Decimal('3'), Decimal('2')) == '3 * 2 = 6'
+assert calculator.division(Decimal('3'), Decimal('2')) == '3 / 2 = 1.5'
+assert calculator.division(Decimal('5'),
+                           Decimal('0')) == 'Error! Division by zero!'
+assert calculator.exponentiation(Decimal('2'), Decimal('3')) == '2 ^ 3 = 8'
```

Comment:
Internally you are using it as a number, so you can use it here as well
```suggestion
assert calculator.summary(3, 2) == '3 + 2 = 5'
assert calculator.subtraction(5, 2) == '5 - 2 = 3'
assert calculator.multiplication(3, 2) == '3 * 2 = 6'
assert calculator.division(3, 2) == '3 / 2 = 1.5'
assert calculator.division(5, 0) == 'Error! Division by zero!'
assert calculator.exponentiation(2, 3) == '2 ^ 3 = 8'
```

Reply:
You have implemented tests for all internal methods, which is really nice!